mlir/utils/performance/ck-benchmark-driver/ck-atten-benchmark-driver.cpp

```diff
@@ -0,0 +1,498 @@
+    const auto kElems = getSize(kLengths);
+    const auto vElems = getSize(vLengths);
+    const auto oElems = getSize(oLengths);
+    const auto biasElems = getSize(biasLengths);
+
+    const auto qBytes =
+        benchmark::getByteSize(options.elementType, qElems, false);
+    const auto kBytes =
+        benchmark::getByteSize(options.elementType, kElems, false);
+    const auto vBytes =
+        benchmark::getByteSize(options.elementType, vElems, false);
+    const auto oBytes =
+        benchmark::getByteSize(options.elementType, oElems, true);
+    const auto biasBytes =
+        benchmark::getByteSize(options.elementType, biasElems, false);
+
+    qHost = benchmark::allocAndFill(options.elementType, qBytes, false);
+    kHost = benchmark::allocAndFill(options.elementType, kBytes, false);
+    vHost = benchmark::allocAndFill(options.elementType, vBytes, false);
+    oHost = benchmark::allocAndFill(options.elementType, oBytes, true);
+    biasHost = benchmark::allocAndFill(options.elementType, biasBytes, false);
+
+    qDevice = benchmark::getGpuBuffer(qHost, qBytes);
+    kDevice = benchmark::getGpuBuffer(kHost, kBytes);
+    vDevice = benchmark::getGpuBuffer(vHost, vBytes);
+    oDevice = benchmark::getGpuBuffer(oHost, oBytes);
+    biasDevice = benchmark::getGpuBuffer(biasHost, biasBytes);
+  }
+  ~Data() {
+    free(qHost);
+    free(kHost);
+    free(vHost);
+    free(oHost);
+    free(biasHost);
+
+    HIP_ABORT_IF_FAIL(hipFree(qDevice));
+    HIP_ABORT_IF_FAIL(hipFree(kDevice));
+    HIP_ABORT_IF_FAIL(hipFree(vDevice));
+    HIP_ABORT_IF_FAIL(hipFree(oDevice));
+    HIP_ABORT_IF_FAIL(hipFree(biasDevice));
+  }
+
+  void *qHost{nullptr};
+  void *kHost{nullptr};
+  void *vHost{nullptr};
+  void *oHost{nullptr};
+  void *biasHost{nullptr};
+
+  void *qDevice{nullptr};
+  void *kDevice{nullptr};
+  void *vDevice{nullptr};
+  void *oDevice{nullptr};
+  void *biasDevice{nullptr};
+
+  std::vector<ck::index_t> qLengths{};
+  std::vector<ck::index_t> kLengths{};
+  std::vector<ck::index_t> vLengths{};
+  std::vector<ck::index_t> oLengths{};
+  std::vector<ck::index_t> biasLengths{};
+
+  std::vector<ck::index_t> qStrides{};
+  std::vector<ck::index_t> kStrides{};
+  std::vector<ck::index_t> vStrides{};
+  std::vector<ck::index_t> oStrides{};
+  std::vector<ck::index_t> biasStrides{};
+
+private:
+  Options options;
+};
+
+template <typename T>
+std::ostream &operator<<(std::ostream &os, std::vector<T> &v) {
+  for (auto &item : v) {
+    os << item << ", ";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, Data &data) {
+  os << "Q\n\t"
+     << "lengths: " << data.qLengths << "\n\tstrides: " << data.qStrides
+     << "\n";
+  os << "K\n\t"
+     << "lengths: " << data.kLengths << "\n\tstrides: " << data.kStrides
+     << "\n";
+  os << "V\n\t"
+     << "lengths: " << data.vLengths << "\n\tstrides: " << data.vStrides;
+  return os;
+}
+
+double getGemmsFlopCount(const Options &options) {
```

Comment:
nit : is "Gemms" flop count ? 
Please add a comment to say keep this implementation synced with perfRunner.py  + function name